### PR TITLE
fix: Run system tests in docker

### LINKED ISSUE #3436: Less tests are executed after #3427
While debugging another issue, @v1v has noticed that there is a set of tests that has been removed from the CI execution when https://github.com/elastic/apm-server/pull/3427 was merged:
- [build](https://apm-ci.elastic.co/job/apm-server/job/apm-server-mbp/job/master/372) that ran before the above PR was merged.
- [build](https://apm-ci.elastic.co/job/apm-server/job/apm-server-mbp/job/master/373) that ran after the above PR was merged.

For instance:

image

vs

image

There are 159 tests that have been skipped:

image

We'd like to make those tests run again

### PR DESCRIPTION
**Pull request by mdelapenya**
## Motivation/summary
Bring back system tests in Docker execution, as because of #3427, the `SYSTEM_TEST_TARGET` environment variable was not populated in the Makefile causing the tests not to be run.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
- [x] My code follows the style guidelines of this project (run `make check-full` for static code checks and linting)
- [x] I have rebased my changes on top of the latest master branch

- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally with my changes

- [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

## How to test these changes

Please run tests as in the CI with `make update docker-system-tests`

## Related issues

- Relates #3427
- Closes #3436

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -80,7 +80,7 @@ system-tests: $(PYTHON_BIN) apm-server.test
 .PHONY: docker-system-tests
 docker-system-tests: docker-compose.override.yml
 	docker-compose build
-	docker-compose run --rm -T beat make system-tests
+	SYSTEM_TEST_TARGET=$(SYSTEM_TEST_TARGET) docker-compose run --rm -T beat make system-tests
```
Comment (axw) [inline Makefile]:
```suggestion
	docker-compose run --rm -T beat make system-tests
```

**Comment by axw** [comment]
Hmm, seems to fail in CI. It worked for me :thinking:

File: Makefile [hunk 2]
```diff
@@ -78,6 +78,7 @@ system-tests: $(PYTHON_BIN) apm-server.test
 	INTEGRATION_TESTS=1 TZ=UTC $(PYTHON_BIN)/nosetests $(NOSETESTS_OPTIONS) $(SYSTEM_TEST_TARGET)
 
 .PHONY: docker-system-tests
+docker-system-tests: export SYSTEM_TEST_TARGET=$(SYSTEM_TEST_TARGET)
```
Comment (axw) [inline Makefile]: sorry, my mistake: this should be `export SYSTEM_TEST_TARGET:=$(SYSTEM_TEST_TARGET)` (note the colon)

Comment (mdelapenya) [reply Makefile]: I'll update the PR ;)

**Pull request merged**